fix(linter/no-img-element): improve diagnostic and docs

DonIsaac opened the pull request:
## What This PR Does

Updates `eslint-next-plugin/no-img-element` to make diagnostic messages/labels/etc more informative and add documentation

camc314 commented:
### Merge activity

* **May 9, 4:47 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

DonIsaac committed:
fix(linter/no-img-element): improve diagnostic and docs (#10908)

## What This PR Does

Updates `eslint-next-plugin/no-img-element` to make diagnostic messages/labels/etc more informative and add documentation
